phpseclib/Net/SFTP.php

```diff
@@ -2310,9 +2324,14 @@ private function delete_recursive(string $path, int &$i): bool
         $i = 0;
         $entries = $this->readlist($path, true);
 
-        // normally $entries would have at least . and .. but it might not if the directories
-        // permissions didn't allow reading
-        if (empty($entries)) {
+        // The folder does not exist at all, so we cannot delete it.
+        if ($entries === StatusCode::NO_SUCH_FILE) {
+            return false;
+        }
+
+        // Normally $entries would have at least . and .. but it might not if the directories
+        // permissions didn't allow reading. If this happens then default to an empty list of files.
+        if (($entries === false) || is_int($entries)) {
```

Comment:
On line 1565 it's `$entries === false || is_int($entries)` but on this one it's `($entries === false) || is_int($entries)`. They prob ought to be consistent.

Anyway, let me backport this to 1.0 / 2.0 / 3.0 and I can change this while I'm at it.

Not sure if I'll be able to do this today (got some stuff going on in < 1h) but I'll see what I can do!

Reply:
I made line 1565 consistent. Personally I like to have "more" brackets, so that there is no doubt about order of evaluation of stuff. But I don't know what the project preference/policy is.

Reply:
Let me know if I should make PRs to the old branches.

Reply:
> I made line 1565 consistent. Personally I like to have "more" brackets, so that there is no doubt about order of evaluation of stuff. But I don't know what the project preference/policy is.

It'd be more consistent with phpseclib's current coding standards to do `$entries === false || is_int($entries)`. A few examples:

https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Crypt/Common/SymmetricKey.php#L738
https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Crypt/Common/SymmetricKey.php#L811
etc

If we were going to go with more parenthesis then ideally phpseclib would be updated in all the other places where it doesn't do that currently, as well, which would be quite a bit more work.

> Let me know if I should make PRs to the old branches.

Yah - that'd be awesome - thanks!!